presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -83,7 +83,7 @@ public OrcReader(OrcDataSource orcDataSource, MetadataReader metadataReader, Dat
         // figure out the size of the file using the option or filesystem
         long size = orcDataSource.getSize();
         if (size <= 0) {
-            throw new OrcCorruptionException("Malformed ORC file %s. Invalid file size %s", orcDataSource, size);
+            throw new OrcCorruptionException(orcDataSource.getId(), "Invalid file size %s", size);
```

Comment:
You could avoid all the `getId()` calls if you create an interface `OrcDataSourceIdSupplier` and make the constructor take that.